File: HTTPShortcuts/curl_command/src/test/java/ch/rmy/curlcommand/CommandLineBuilderTest.java
package ch.rmy.curlcommand;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class CommandLineBuilderTest {

    @Test
    public void testCommandLineBuilder() {
        String expected = "foo -bla 25 --foo-bar \"Hello \\\"World\\\"\" true -bla Hello \"\\\"World\\\"\"";
        String actual = new CommandLineBuilder("foo")
                .option("-bla")
                .argument(25)
                .option("--foo-bar")
                .argument("Hello \"World\"")
                .argument(true)
                .option("-bla", "Hello", "\"World\"")
                .build();
        assertEquals(expected, actual);
    }

    @Test
    public void testBackslashes() {
        String expected = "foo \" \\\" \\";
        String actual = new CommandLineBuilder("foo")
                .argument(" \\")
                .argument("\\")
                .build();
        assertEquals(expected, actual);
    }

    @Test(expected = IllegalArgumentException.class)
    public void testIllegalOption() {
        new CommandLineBuilder("foo")
                .option("invalidOption");
    }

}

File: HTTPShortcuts/curl_command/src/test/kotlin/ch/rmy/curlcommand/CommandLineBuilderTest.kt
package ch.rmy.curlcommand

import kotlin.test.Test
import kotlin.test.assertEquals
import org.junit.jupiter.api.assertThrows

class CommandLineBuilderTest {

    @Test
    fun testCommandLineBuilder() {
        val expected = "foo -bla 25 --foo-bar \"Hello \\\"World\\\"\" true -bla Hello \"\\\"World\\\"\""
        val actual = CommandLineBuilder("foo")
            .option("-bla")
            .argument(25)
            .option("--foo-bar")
            .argument("Hello \"World\"")
            .argument(true)
            .option("-bla", "Hello", "\"World\"")
            .build()
        assertEquals(expected, actual)
    }

    @Test
    fun testBackslashes() {
        val expected = "foo \" \\\" \\"
        val actual = CommandLineBuilder("foo")
            .argument(" \\")
            .argument("\\")
            .build()
        assertEquals(expected, actual)
    }

    @Test
    fun testIllegalOption() {
        assertThrows<IllegalArgumentException> {
            CommandLineBuilder("foo")
                .option("invalidOption")
        }
    }
}
